src/main/scala/com/github/unchama/generic/effect/concurrent/Retry.scala

```diff
@@ -0,0 +1,20 @@
+package com.github.unchama.generic.effect.concurrent
```

Comment:
```suggestion
package com.github.unchama.generic.effect
```

下に `MonadThrowExtra` って名前で作ったほうが良さそうです（`: MonadThrow` 制約は、例えば `SyncIO` の様な concurrent でない effect にも通用するので）